#!/usr/bin/env python

PACKAGE = "lidar_fusion_dynamic"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 定义lidar1的参数
gen.add("lidar1_x", double_t, 0, "Lidar 1 X", 0.0, -10.0, 10.0)
gen.add("lidar1_y", double_t, 0, "Lidar 1 Y", 0.0, -10.0, 10.0)
gen.add("lidar1_z", double_t, 0, "Lidar 1 Z", 0.0, -10.0, 10.0)
gen.add("lidar1_roll", double_t, 0, "Lidar 1 Roll", 0.0, -3.14, 3.14)
gen.add("lidar1_pitch", double_t, 0, "Lidar 1 Pitch", 0.0, -3.14, 3.14)
gen.add("lidar1_yaw", double_t, 0, "Lidar 1 Yaw", 0.0, -3.14, 3.14)

# 定义lidar2的参数
gen.add("lidar2_x", double_t, 0, "Lidar 2 X", 0.0, -10.0, 10.0)
gen.add("lidar2_y", double_t, 0, "Lidar 2 Y", 0.0, -10.0, 10.0)
gen.add("lidar2_z", double_t, 0, "Lidar 2 Z", 0.0, -10.0, 10.0)
gen.add("lidar2_roll", double_t, 0, "Lidar 2 Roll", 0.0, -3.14, 3.14)
gen.add("lidar2_pitch", double_t, 0, "Lidar 2 Pitch", 0.0, -3.14, 3.14)
gen.add("lidar2_yaw", double_t, 0, "Lidar 2 Yaw", 0.0, -3.14, 3.14)

gen.generate(PACKAGE, "lidar_fusion_dynamic", "LidarFusionConfig")
